front/src/applications/stdcm/components/StdcmForm/StdcmConfig.tsx

```diff
@@ -180,6 +207,53 @@ const StdcmConfig = ({
     };
   }, [showMessage, formErrors]);
 
+  useEffect(() => {
+    if (
+      formErrors?.errorType !== StdcmConfigErrorTypes.MISSING_INFORMATIONS ||
+      !formErrors.errorDetails?.missingFields
+    ) {
+      return;
+    }
+
+    const updatedMissingFields = filterMissingFields({
+      missingFields: formErrors.errorDetails.missingFields,
+      rollingStock,
+      totalMass,
+      totalLength,
+      maxSpeed,
+      origin,
+      destination,
+    });
+
+    const previousFields = formErrors.errorDetails.missingFields;
+    const hasChanged =
+      previousFields.length !== updatedMissingFields.length ||
+      previousFields.some((field, i) => field !== updatedMissingFields[i]);
```

Comment:
Do we need the first check ?
```suggestion
    const hasChanged = previousFields.some((field, i) => field !== updatedMissingFields[i]);
```

Reply:
`some()`only compares items at the same indexes, so if a new field is added at the end of the list, it won’t detect any difference.